### PR TITLE
Update Compute*KZGProof in java bindings

### PR DESCRIPTION
**Pull request by jtraglia**
This PR updates the Java bindings to be in accordance with:

* https://github.com/ethereum/c-kzg-4844/pull/174

### CODE REVIEW

File: bindings/java/src/main/java/ethereum/ckzg4844/CKZG4844JNI.java [hunk 1]
```diff
@@ -156,19 +156,20 @@ public static <T> void loadTrustedSetupFromResource(String resource, Class<T> cl
...
    */
-  public static native byte[] computeKzgProof(byte[] blob, byte[] z_bytes);
+  public static native Tuple computeKzgProof(byte[] blob, byte[] z_bytes);
 
   /**
    * Given a blob, return the KZG proof that is used to verify it against the commitment
    *
    * @param blob blob bytes
+   * @param commitment_bytes commitment bytes
    * @return the aggregated proof
```
Comment (StefanBratanov) [inline bindings/java/src/main/java/ethereum/ckzg4844/CKZG4844JNI.java:169]: just a nit, should this be just `the proof`

File: bindings/java/src/main/java/ethereum/ckzg4844/CKZG4844JNI.java [hunk 2]
```diff
@@ -156,19 +156,20 @@ public static <T> void loadTrustedSetupFromResource(String resource, Class<T> cl
    *
    * @param blob blob bytes
    * @param z_bytes a point
-   * @return the proof
+   * @return A tuple of proof and y
```
Comment (StefanBratanov) [inline bindings/java/src/main/java/ethereum/ckzg4844/CKZG4844JNI.java:159]: For more clarity, I would change it to A tuple of proof and value y = f(z)

**Comment by StefanBratanov** [comment]
LGTM

File: bindings/java/c_kzg_4844_jni.c [hunk 3]
```diff
@@ -242,14 +267,23 @@ JNIEXPORT jbyteArray JNICALL Java_ethereum_ckzg4844_CKZG4844JNI_computeBlobKzgPr
...
+  Bytes48 *commitment_native = (Bytes48 *)(*env)->GetByteArrayElements(env, commitment_bytes, NULL);
 
   jbyteArray proof = (*env)->NewByteArray(env, BYTES_PER_PROOF);
   KZGProof *proof_native = (KZGProof *)(uint8_t *)(*env)->GetByteArrayElements(env, proof, NULL);
 
-  C_KZG_RET ret = compute_blob_kzg_proof(proof_native, blob_native, settings);
+  C_KZG_RET ret = compute_blob_kzg_proof(proof_native, blob_native, commitment_native, settings);
 
   (*env)->ReleaseByteArrayElements(env, blob, (jbyte *)blob_native, JNI_ABORT);
+  (*env)->ReleaseByteArrayElements(env, commitment_bytes, (jbyte *)commitment_native, JNI_ABORT);
```
Comment (asn-d6) [inline bindings/java/c_kzg_4844_jni.c:286]: Do we care that in some places ReleaseByteArrayElements does `mode == JNI_ABORT` and in others `mode == 0`?

File: bindings/java/c_kzg_4844_jni.c [hunk 4]
```diff
@@ -211,23 +211,48 @@ JNIEXPORT jbyteArray JNICALL Java_ethereum_ckzg4844_CKZG4844JNI_computeKzgProof(
...
   {
     throw_c_kzg_exception(env, ret, "There was an error in computeKzgProof.");
     return NULL;
   }
 
-  return proof;
+  jclass tupleClass = (*env)->FindClass(env, "ethereum/ckzg4844/Tuple");
+  if (tupleClass == NULL)
+  {
+    throw_c_kzg_exception(env, ret, "Failed to find Tuple class.");
```
Comment (jtraglia) [inline bindings/java/c_kzg_4844_jni.c:234]: Oh, these should probably be changed from `ret` to `C_KZG_ERROR`. Will fix.

Comment (jtraglia) [reply bindings/java/c_kzg_4844_jni.c]: Thanks, I like that better. Updated.

**Comment by jtraglia** [comment]
I think this one is good. Going to merge.

**Pull request merged**